### DISCUSSION TITLE
Have there been changes to auto dark/light mode?

### DISCUSSION THREAD

**Discussion by sagarbehere** [razonyang/hugo-theme-bootstrap#530] [Q&A] [answered]
I was using v0.67.1 until recently and am pretty sure that, at least on mobile, my site would automatically switch between light and dark modes based on the time of the day. 

After recently upgrading to v0.79.1, the auto switching is no longer happening. Have there been any changes to this? (I did try clearing all browser cookies.)

**Comment by razonyang** [comment]
No, it will detect the mode that the device used(tested on my phone), please check your device settings, so that the device can switch mode base on the local time automatically.

**Comment by sagarbehere** [comment]
Hmm, I’m not seeing the automatic switch to dark mode :-( My iOS device is set to switch to dark mode automatically, from sunset to sunrise (local time zone). Any idea how to debug this?

Maybe you can test this too? Does [https://sagar.se](https://sagar.se) show up in dark mode on your device at night?

**Comment by sagarbehere** [comment]
Also, [https://hbs.razonyang.com/en/](https://hbs.razonyang.com/en/) is not showing up in dark mode on my device either. It only switches to dark mode when I change it manually from the site’s sidebar on the right.

**Comment by razonyang** [comment]
@sagarbehere I'll recheck this tonight

**Comment by razonyang** [comment]
@sagarbehere I tested your site on my phone(Huawei Mate), works as expected. I don't have any iPhone, so I'm not sure if this is just a problem on the iPhone. It would be great if you can provide some screen recording.

**Comment by sagarbehere** [comment, marked as answer]
Hmm... I removed all browsing history/cache/saved data etc. from my browser and tried again. Everything is working now. So the problem has disappeared `¯\_(ツ)_/¯`

I'm curious how this feature works. If a user manually chooses a light/dark setting from the right sidebar, is that choice "remembered" for  future visits? If yes, how can users reset their preference from a manual choice back to automatic switching based on their device settings?

**Reply by razonyang** [reply]
system selection will change user selection, it's the weakness of current implementation, I'll bring a new implementation with three options for users very soon.
